rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/statistics/ProbablisticSelectionManager.java

```diff
@@ -277,7 +277,13 @@ private SelectionScript replaceBindings(SelectionScript script, Map<String, Seri
         String scriptContent = script.getScript();
         if (bindings != null) {
             for (Map.Entry<String, Serializable> entry : bindings.entrySet()) {
-                scriptContent = scriptContent.replace(entry.getKey(), entry.getValue().toString());
+                String reservedKeyword = entry.getKey();
+                Serializable binding = entry.getValue();
+                if (binding instanceof Map) {
+                    scriptContent = replaceBindingKeysByTheirValue(scriptContent, (Map<String, Serializable>) binding);
+                } else {
+                    scriptContent = scriptContent.replace(reservedKeyword, binding.toString());
```

Comment:
please add a defensive != null check, if binding == null, this will result in a NPE

Reply:
Done.